maven-plugins/stager-maven-plugin/src/main/java/io/helidon/build/maven/stager/StagerMojo.java

```diff
@@ -455,5 +468,26 @@ public void logError(String msg, Object... args) {
         public void logDebug(String msg, Object... args) {
             log.debug(String.format(msg, args));
         }
+
+        @Override
+        public void submit(Callable<CompletionStage<Void>> task) {
+            tasksQueue.add(task);
+        }
+
+        @Override
+        public void awaitTermination() {
+            try {
```

Comment:
What if a task invokes `submit()` as part of its execution ?
We probably need something like this:

```java
Collection<Callable<CompletionStage<Void>>> tasks;
synchronized (tasksQueue) {
    tasks = new LinkedList<>(tasksQueue);
    tasksQueue.clear();
}
executor.invokeAll(tasks);
```

Another way would be to use something like this:
```java
ArrayBlockingQueue<Callable<CompletionStage<Void>>> tasksQueue; // declaration
List<Callable<CompletionStage<Void>>> tasks = new LinkedList<>()
tasksQueue.drainTo(tasks);
executor.invokeAll(tasks);
```

If using `ArrayBlockingQueue`, then `submit` does not need to use a synchronized block.